rotkehlchen/chain/ethereum/modules/defisaver/decoder.py

```diff
@@ -64,10 +141,51 @@ def _decode_substorage_action(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_flashloan_events(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes DeFi Saver events from the given context.
+        """
+        # We want only to remap the ERC20 transfers, so
+        # we just match the counterparty for triggering the post-decoding rules
+        for event in context.decoded_events:
+            if (
+                event.address in self.flashloan_addresses and
+                event.location_label is not None and
+                self.base.is_tracked(string_to_evm_address(event.location_label))
+            ):
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_subtype = HistoryEventSubType.GENERATE_DEBT  # TODO: should  we create a new subtype for flashloans?  # noqa: E501
+                    event.counterparty = CPT_DEFISAVER
+                    event.notes = f'Executed flashloan of {event.balance.amount} {event.asset.symbol_or_name()} via DefiSaver'  # noqa: E501
```

Comment:
In rotki's events we stick to present tense. Also I would change execute to take out a flashloan.
```suggestion
                    event.notes = f'Take out a flashloan of {event.balance.amount} {event.asset.symbol_or_name()} via DefiSaver'  # noqa: E501
```